tests/testthat/test-independent-testing-prepare_ae_summary.R

```diff
@@ -0,0 +1,639 @@
+    label = "serious drug-related adverse events resulting in discontinuation"
+  )
+
+
+meta <- meta %>%
+  define_analysis(
+    name = "ae_summary",
+    title = "Adverse Event Summary"
+  )
+
+
+meta <- meta %>% meta_build()
+
+meta$plan <- meta$plan %>%
+  mutate(output_report = spec_filename(meta))
+
+test_meta <- meta
+# usethis::use_data(test_meta, overwrite = TRUE)
+
+x <- prepare_ae_summary(
+  meta = test_meta,
+  population = "apat",
+  observation = "trtemfl",
+  parameter = sum_par
+)
+
+xx <- tibble(cbind(x$name, x$n)) %>% mutate(x = row_number())
+xxp <-x$prop %>% mutate(x = row_number())
+
+adsl_tot <- adsl %>% mutate(TRT01AN = 99) # add total rows into calculation
+adsl_tot <- rbind(adsl, adsl_tot)
+
+res_tot <- adsl_tot %>%
+  group_by(TRT01AN, ITTFL) %>%
+  summarise(n = n_distinct(USUBJID), .groups = 'drop')
+
+res_tot1 <- res_tot %>% mutate(z=n) %>% select(-c(n))
+
+
+test_that("population count", {
+
+  res <- res_tot %>%
+    mutate(x = paste0("n_", row_number()))
+
+  # res_total <- adsl_tot %>%
+  #   group_by(TRT01A, ITTFL) %>%
+  #   summarise(tot = n_distinct(USUBJID)) #total within each grp
+
+
+  # res <- res %>%
+  #   left_join(res_total, by = "TRT01A", "ITTFL") %>%
+  #   mutate(pct = formatC(100*n/tot, digits = 0, format = "f", flag = "0")) #calculate pct
+  #
+  # dum_pct <- formatC(0,digits = decimal, format = "f", flag = "0")
+
+  res1 <- res %>%
+    pivot_wider(
+      id_cols = "ITTFL",
+      names_from = "x",
+      values_from = n,
+      values_fill = list(n = 0)
+    )
+
+  expect_equal(xx %>% filter(x == 1) %>% select(-c("x$name", x)), res1 %>% select(-c(ITTFL)))
+  expect_equal(xx %>% filter(x == 1) %>% select(-c("x$name", x)), tibble(x$n_pop))
+})
+
+test_that("Parameter 'any' count", {
+  adae_itt <- full_join(adsl %>% select(USUBJID, TRT01AN, ITTFL), # merge with adsl to get percentage
+    adae,
+    by = "USUBJID"
+  ) %>% subset(ITTFL == "Y" & TRTEMFL == "Y")
+
+  adae_tot <- adae_itt %>% mutate(TRT01AN = 99) # add total rows into calculation
+  adae_tot <- rbind(adae_itt, adae_tot)
+
+  res_ae <- adae_tot %>%
+    group_by(TRT01AN, ITTFL) %>%
+    summarise(n = n_distinct(USUBJID), .groups = 'drop')
+
+  res_ae <- full_join(res_ae,res_tot1, by = c("TRT01AN", "ITTFL")) %>%
+    mutate(pct = formatC(100*n/z, digits = 13, format = "f", flag = "0"))
+
+  res_ae <- res_ae %>%
+    mutate(x = paste0("n_", row_number()),
+           prop=paste0("prop_", row_number()))
+
+  res_ae2 <- res_ae %>%
+    pivot_wider(
+      id_cols = "ITTFL",
+      names_from = "x",
+      values_from = n,
+      values_fill = list(n = 0)
+    )
+
+  res_prop2 <- res_ae %>%
+    pivot_wider(
+      id_cols = "ITTFL",
+      names_from = "prop",
+      values_from = pct,
+      values_fill = list(n = 0)
+    )
+  expect_equal(xx %>% filter(x == 2) %>% select(-c("x$name", x)), res_ae2 %>% select(-c(ITTFL)))
+  expect_equal(xxp %>% filter(x == 2) %>% select(-c("x$name", x)), res_prop2 %>% select(-c(ITTFL)))
```

Comment:
Hi @ginnaram13, please check if `xxp` should be `xx`. Since there is no column called `"x$name"` in `xxp`. I also commit some new tests based on your codes. Please pull before you make a new commit/push.

Reply:
> Hi @ginnaram13, please check if `xxp` should be `xx`. Since there is no column called `"x$name"` in `xxp`. I also commit some new tests based on your codes. Please pull before you make a new commit/push.

Hi @LittleBeannie , This is not my final code. I am not getting metalite.ae updates after pull.